src/main.rs

```diff
@@ -4,44 +4,72 @@ use quicksilver::{
     Input, Result, Settings, Window,
 };
 
-use rand::distributions::{Distribution, Uniform};
+use rand::distributions::{Distribution, Uniform, WeightedIndex};
 use std::vec::Vec;
+use rand::prelude::ThreadRng;
 
+#[derive(Debug)]
 struct Node {
     x: f32,
     y: f32,
 }
 
+const HEIGHT: f32 = 1024.0;
+const LENGTH: f32 = 768.0;
+
+impl Node {
+    pub fn gen(mut rng: ThreadRng) -> Self {
+        let x_dist = Uniform::from(0.0..HEIGHT);
+        let y_dist = Uniform::from(0.0..LENGTH);
+        Node {
+            x: x_dist.sample(&mut rng),
+            y: y_dist.sample(&mut rng),
+        }
+    }
+}
+
 struct World {
     nodes: Vec<Node>,
     connections: Vec<(u32, u32)>,
 }
 
-fn gen_world() -> World {
-    let node_count = 20;
+impl World {
+    pub fn gen() -> World {
+        let node_count = 20;
 
-    let mut nodes = Vec::new();
+        let mut rng = rand::thread_rng();
+        let nodes: Vec<Node> = (0..node_count)
+            .map(|_| Node::gen(rng))
+            .collect();
 
-    let mut rng = rand::thread_rng();
-    let x_dist = Uniform::from(0.0..1024.0);
-    let y_dist = Uniform::from(0.0..768.0);
-    for _ in 0..node_count {
-        let new_node = Node {
-            x: x_dist.sample(&mut rng),
-            y: y_dist.sample(&mut rng),
+        let connections_count = 40;
+        let mut connections = Vec::new();
+        for _ in 0..connections_count {
+            let distrib = Uniform::from(0..nodes.len());
+            let node_index = distrib.sample(&mut rng);
+            let node = nodes.get(node_index).unwrap();
+            let weights: Vec<f32> = nodes.iter()
+                .map(|i| euclidean_dist(node, i))
+                .collect();
+            // max from sum
+            // let max_weights: f32 = weights.iter().sum();
+            // max fron biggest element
+            let max_weights: f32 = weights.iter().fold(0.0f32, |val, next_val| next_val.max(val));
+            let normalized_weights: Vec<f32> = weights.into_iter()
+                .map(|elem| 1.0 - (elem / max_weights))
+                .collect();
+            let weigthed_distrib = WeightedIndex::new(&normalized_weights).unwrap();
+            connections.push((node_index as u32, weigthed_distrib.sample(&mut rng) as u32))
         };
 
-        nodes.push(new_node);
-    }
 
-    let connections_count = 40;
-    let mut connections = Vec::new();
-    let conn_dist = Uniform::from(0..node_count);
-    for _ in 0..connections_count {
-        connections.push((conn_dist.sample(&mut rng), conn_dist.sample(&mut rng)))
+
+        World { nodes, connections }
     }
+}
 
-    World { nodes, connections }
+fn euclidean_dist(n1: &Node, n2: &Node) -> f32 {
+    ((n1.x - n2.x).powi(2) + (n1.y - n2.y).powi(2)).sqrt()
```

Comment:
just for fun - sqrt is a monotonic function and for weights we can use just `(n1.x - n2.x).powi(2) + (n1.y - n2.y).powi(2)`

Reply:
do not fix it :)